### ISSUE TITLE
Adding a third axis on my PCA plot? 

### ISSUE THREAD

**Issue by MeganS92** [satijalab/seurat#742]
Hi all, 

I have just started using Seurat a few weeks ago, And currently, we have a query we would like to solve using my dataset. 

Is there a way of plotting my PCA plots (eg PC1 and PC2) against the number of reads, in my Seurat object? I have researched online, and it sees like one can perform this on a GGPLOT2 function using sec_axis (https://rpubs.com/MarkusLoew/226759). 

However, is there a way of incorporating this on Seurat? This is because my RNA-seq object is built in Seurat (someone else has built this), and I don't know how to actually pull things out like number of reads/ counts out of the Seurat object, and incorporate it together with my TPM matrix, to perform this on a basic GGPLOT2. 

This is my example code (without the third axis): 

PCAplot(object, dim.1=1, dim.2=2, group.by="timepoint", cols.use= NULL)

If anyone could help this, I would be grateful! Sorry if I have asked a repeated question, I did try searching around before posting this. 

Regards,
Megan.

**Comment by leonfodoulian** [comment]
Hi Megan,

To check whether any of the PCs is correlated with `nGene` or `nUMI` (data located in the `object @meta.data` containing the number of genes and UMI for each cell, respectively), you can use the `Seurat::GenePlot()` function. This will plot `nGene`/`nUMI` as a function of each PC separately:

```R
GenePlot(object = object,
         gene1 = "PC1", # change to "PC2", etc. to plot other components
         gene2 = "nGene", # change to "nUMI", etc.
         cex.use = 1,
         pch.use = 19)
```

Best,
Leon

**Comment by MeganS92** [comment]
Hey Leon, 

Thanks so much for this. This is very useful (also for the prompt reply). 

I guess the only limitation to that is that with my dataset, 1 PC alone doesn't allow me to separate out all my "clusters", and so it may still be hard to visualise if any of my clusters are affected by say, number of genes. I will still try this though. I was wondering if there are other functions to add a 3rd parameter within Seurat. 

Thanks still! Much appreciated. 

Kind regards,
Megan.

**Comment by MeganS92** [comment]
Hi Leon, 

Sorry to add on another question, 

I ran my code, which is simply: 

GenePlot(object, gene1= "PC1", gene2="nGene) and it produced a plot of correlation for me, but I am very confused over the colour scheme... What does each colour represent? My samples contain samples from different timepoints. Is there a way of assigning colours based on that? 

Thanks,
Megan.

**Comment by leonfodoulian** [comment]
> I guess the only limitation to that is that with my dataset, 1 PC alone doesn't allow me to separate out all my "clusters", and so it may still be hard to visualise if any of my clusters are affected by say, number of genes.

To visualise if a given cluster has more expressed genes or UMI, simply use: 

```R
Seurat::FeaturePlot(object = object,
                    features.plot = "nGene")
```

PCs are uncorrelated components, and I'm not sure that what you want to do is at all informative (I might be wrong!). One approach for this would be to identify all PCs that are correlated with `nGene` or `nUMI`, and then use `Seurat::PCAPlot()` and see if any cluster or timepoint is separate by these PCs.

> What does each colour represent?

Colours represent cluster IDs. If you haven't computed `Seurat::FindClusters()` yet, colours represent cell IDs in `object@meta.data$orig.ident`.

> My samples contain samples from different timepoints. Is there a way of assigning colours based on that?

You can use the following code to colour cells by timepoints:

```R
object <- Seurat::SetAllIdent(object = object, id = "timepoints") # set cell ID to timepoints

Seurat::GenePlot(object = object,
                 gene1 = "PC1", # change to "PC2", etc. to plot other components
                 gene2 = "nGene", # change to "nUMI", etc.
                 cex.use = 1,
                 pch.use = 19)
```

Best,
Leon

**Comment by MeganS92** [comment]
Hi Leon, 

Thanks for patiently explaining my questions. 

I have done the visualisation using Feature Plot, and Gene Plot, and I guess this would suffice for now. Thanks also for the handy tip in assigning cluster ID colours based on timepoints. 

I think what I was originally asking is a little too complicated, and your solutions are sufficient for now. 

Thanks again, I will close this issue now. 

Regards,
Megan.

**Comment by MeganS92** [comment]
Closes: #742
